src/transformers/onnx/__main__.py

```diff
@@ -47,10 +47,16 @@ def main():
         args.output.parent.mkdir(parents=True)
 
     # Allocate the model
-    tokenizer = AutoTokenizer.from_pretrained(args.model)
     model = FeaturesManager.get_model_from_feature(args.feature, args.model)
     model_kind, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=args.feature)
     onnx_config = model_onnx_config(model.config)
+    # Check the modality of the inputs and instantiate the appropriate preprocessor
+    if model.main_input_name == "input_ids":
+        preprocessor = AutoTokenizer.from_pretrained(args.model)
+    elif model.main_input_name == "pixel_values":
+        preprocessor = AutoFeatureExtractor.from_pretrained(args.model)
```

Comment:
As discussed offline, there is a better way to instantiate the right preprocessor.